### PR TITLE
New go at Outlaw t21 improvements

### PR DESCRIPTION
**Pull request by weakweak**

### CODE REVIEW

File: engine/class_modules/sc_rogue.cpp [hunk 1]
```diff
@@ -7389,13 +7391,13 @@ void rogue_t::init_action_list()
     cds -> add_talent( this, "Cannonball Barrage", "if=spell_targets.cannonball_barrage>=1" );
     cds -> add_action( this, "Adrenaline Rush", "if=!buff.adrenaline_rush.up&energy.deficit>0" );
     cds -> add_talent( this, "Marked for Death", "target_if=min:target.time_to_die,if=target.time_to_die<combo_points.deficit|((raid_event.adds.in>40|buff.true_bearing.remains>15-buff.adrenaline_rush.up*5)&!stealthed.rogue&combo_points.deficit>=cp_max_spend-1)" );
-    cds -> add_action( this, "Sprint", "if=equipped.thraxis_tricksy_treads&!variable.ss_useable" );
+    cds -> add_action( this, "Sprint", "if=!talent.death_from_above.enabled&equipped.thraxis_tricksy_treads&!variable.ss_useable&" );
```
Comment (aethys256) [inline engine/class_modules/sc_rogue.cpp]: You forgot a trailing `&`, making an error =)

File: engine/class_modules/sc_rogue.cpp [hunk 2]
```diff
@@ -7389,13 +7391,13 @@ void rogue_t::init_action_list()
...
     cds -> add_action( "darkflight,if=equipped.thraxis_tricksy_treads&!variable.ss_useable&buff.sprint.down" );
     cds -> add_action( this, "Curse of the Dreadblades", "if=combo_points.deficit>=4&(!talent.ghostly_strike.enabled|debuff.ghostly_strike.up)" );
 
     // Finishers
     action_priority_list_t* finish = get_action_priority_list( "finish", "Finishers" );
-    finish -> add_action( this, "Between the Eyes", "if=(mantle_duration>=0.2&!equipped.thraxis_tricksy_treads)|(equipped.greenskins_waterlogged_wristcuffs&!buff.greenskins_waterlogged_wristcuffs.up)" );
```
Comment (aethys256) [inline engine/class_modules/sc_rogue.cpp]: BtE w/ Mantle isn't worth anymore ?

File: engine/class_modules/sc_rogue.cpp [hunk 3]
```diff
@@ -7332,14 +7332,16 @@ void rogue_t::init_action_list()
     precombat -> add_action( this, "Curse of the Dreadblades", "if=combo_points.deficit>=4" );
 
     // Main Rotation
-    def -> add_action( "variable,name=rtb_reroll,value=!talent.slice_and_dice.enabled&buff.loaded_dice.up&(rtb_buffs<2|rtb_buffs=2&!buff.true_bearing.up)", "Fish for '2 Buffs' when Loaded Dice is up. Also try to get TB with Loaded Dice and 2 other buffs up. With SnD, consider that we never have to reroll." );
-    def -> add_action( "variable,name=ss_useable_noreroll,value=(combo_points<5+talent.deeper_stratagem.enabled-(buff.broadsides.up|buff.jolly_roger.up)-(talent.alacrity.enabled&buff.alacrity.stack<=4))", "Condition to use Saber Slash when not rerolling RtB or when using SnD" );
+    def -> add_action( "variable,name=rtb_reroll,value=!talent.slice_and_dice.enabled&buff.loaded_dice.up&(rtb_buffs<2|(rtb_buffs<4&!buff.true_bearing.up))", "Fish for '2 Buffs' when Loaded Dice is up. Also try to get TB with Loaded Dice and 2 other buffs up. With SnD, consider that we never have to reroll." );
```
Comment (aethys256) [inline engine/class_modules/sc_rogue.cpp]: You forgot to update the comment part ;)

**Review by aethys256 — CHANGES_REQUESTED** [review]
Some things must be fixed =)

File: engine/class_modules/sc_rogue.cpp [hunk 4]
```diff
@@ -7389,13 +7391,13 @@ void rogue_t::init_action_list()
     cds -> add_talent( this, "Cannonball Barrage", "if=spell_targets.cannonball_barrage>=1" );
     cds -> add_action( this, "Adrenaline Rush", "if=!buff.adrenaline_rush.up&energy.deficit>0" );
     cds -> add_talent( this, "Marked for Death", "target_if=min:target.time_to_die,if=target.time_to_die<combo_points.deficit|((raid_event.adds.in>40|buff.true_bearing.remains>15-buff.adrenaline_rush.up*5)&!stealthed.rogue&combo_points.deficit>=cp_max_spend-1)" );
-    cds -> add_action( this, "Sprint", "if=equipped.thraxis_tricksy_treads&!variable.ss_useable" );
+    cds -> add_action( this, "Sprint", "if=!talent.death_from_above.enabled&equipped.thraxis_tricksy_treads&!variable.ss_useable" );
     cds -> add_action( "darkflight,if=equipped.thraxis_tricksy_treads&!variable.ss_useable&buff.sprint.down" );
-    cds -> add_action( this, "Curse of the Dreadblades", "if=combo_points.deficit>=4&(!talent.ghostly_strike.enabled|debuff.ghostly_strike.up)" );
+    cds -> add_action( this, "Curse of the Dreadblades", "if=(combo_points.deficit>=4&buff.true_bearing.up)|(combo_points.deficit>=4&(buff.adrenaline_rush.up|time_to_die<20))" );
```
Comment (Mystler) [inline engine/class_modules/sc_rogue.cpp]: Have to nitpick about style in this one. 😛 This should do the same while being much more readable if I am not mistaken.
```if=combo_points.deficit>=4&(buff.true_bearing.up|buff.adrenaline_rush.up|time_to_die<20)```

**Review by Mystler — CHANGES_REQUESTED** [review]

**Review by Mystler — APPROVED** [review]

**Review by aethys256 — APPROVED** [review]

**Pull request merged**